client/app/containers/ConnectorManagement.jsx

```diff
@@ -0,0 +1,216 @@
+    this.setState({ submitting: true })
+    this.store.updateAccount(id, idx, values)
+      .then(fetchedConnector => {
+        this.setState({ submitting: false })
+        if (fetchedConnector.importErrorMessage) {
+          this.setState({ error: fetchedConnector.importErrorMessage })
+        } else {
+          Notifier.info(t('my_accounts account config success'))
+        }
+      })
+      .catch(error => { // eslint-disable-line
+        this.setState({ submitting: false })
+        Notifier.error(t('my_accounts account config error'))
+      })
+  }
+
+  synchronize () {
+    const id = this.state.connector.id
+    const { t } = this.context
+    this.setState({ synching: true })
+    this.store.synchronize(id)
+      .then(fetchedConnector => {
+        this.setState({ synching: false })
+        if (fetchedConnector.importErrorMessage) {
+          this.setState({ error: fetchedConnector.importErrorMessage })
+        }
+      })
+      .catch(error => { // eslint-disable-line
+        this.setState({ synching: false })
+        Notifier.error(t('my_accounts account config error'))
+      })
+  }
+
+  deleteAccount (idx) {
+    const id = this.state.connector.id
+    const { t } = this.context
+    this.setState({ deleting: true })
+    this.store.deleteAccount(id, idx)
+      .then(() => {
+        this.setState({ deleting: false })
+        if (this.state.connector.accounts.length === 0) {
+          this.gotoParent()
+        } else {
+          this.selectAccount(0)
+        }
+        Notifier.info(t('my_accounts account delete success'))
+      })
+      .catch(error => { // eslint-disable-line
+        this.setState({ deleting: false })
+        Notifier.error(t('my_accounts account delete error'))
+      })
+  }
+
+  getDefaultValues () {
+    let defaults = {}
+    Object.keys(this.state.fields).map(k => {
+      defaults[k] = this.state.fields[k].default || ''
+    })
+    return defaults
+  }
+
+  // Set default values for advanced fields that will not be shown
+  // on the initial connection form
+  configureFields (fields, t, connectorName) {
+    if (fields.calendar && !fields.calendar.default) {
+      fields.calendar.default = connectorName
+    }
+    if (fields.folderPath && !fields.folderPath.default) {
+      fields.folderPath.default = t('my_accounts title') + '/' + connectorName
+    }
+    if (fields.folderPath && !fields.folderPath.options) {
+      fields.folderPath.options = this.store.folders.map(f => f.path + '/' + f.name)
+      fields.folderPath.folders = this.store.folders
+    }
+    if (!fields.frequency) {
+      fields.frequency = {
+        type: 'text',
+        advanced: true
+      }
+    }
+    if (fields.frequency && !fields.frequency.default) {
+      fields.frequency.default = 'week'
+    }
+    if (fields.frequency &&
+      (!fields.frequency.options || !fields.frequency.options.length)) {
+      fields.frequency.options = ['hour', 'day', 'week', 'month']
```

Comment:
So we can remove this part to remove confusion about default values of the frequency field here, what do you think?